Plugins/GRPCProtobufGenerator/Plugin.swift

```diff
@@ -0,0 +1,278 @@
+///   - protoDirectoryPath: The root path to the source `.proto` files used as the reference for relative path naming schemes.
+///   - protocPath: The path to `protoc`
+///   - protocGenGRPCSwiftPath: The path to `proto-gen-grpc-swift`.
+/// - Returns: The command to invoke `protoc` with the `proto-gen-grpc-swift` plugin.
+func protocGenGRPCSwiftCommand(
+  inputFile: URL,
+  config: GenerationConfig,
+  protoDirectoryPath: URL,
+  protocPath: URL,
+  protocGenGRPCSwiftPath: URL
+) throws -> PackagePlugin.Command {
+  let outputPathURL = URL(fileURLWithPath: config.outputPath)
+
+  let outputFilePath = deriveOutputFilePath(
+    for: inputFile,
+    protoDirectoryPath: protoDirectoryPath,
+    outputDirectory: outputPathURL,
+    outputExtension: "grpc.swift"
+  )
+
+  let arguments = constructProtocGenGRPCSwiftArguments(
+    config: config,
+    using: config.fileNaming,
+    inputFiles: [inputFile],
+    protoDirectoryPaths: [protoDirectoryPath],
+    protocGenGRPCSwiftPath: protocGenGRPCSwiftPath,
+    outputDirectory: outputPathURL
+  )
+
+  return Command.buildCommand(
+    displayName: "Generating gRPC Swift files for \(inputFile.relativePath)",
+    executable: protocPath,
+    arguments: arguments,
+    inputFiles: [inputFile, protocGenGRPCSwiftPath],
+    outputFiles: [outputFilePath]
+  )
+}
+
+/// Construct the command to invoke `protoc` with the `proto-gen-swift` plugin.
+/// - Parameters:
+///   - inputFile: The input `.proto` file.
+///   - config: The configuration for this operation.
+///   - protoDirectoryPath: The root path to the source `.proto` files used as the reference for relative path naming schemes.
+///   - protocPath: The path to `protoc`
+///   - protocGenSwiftPath: The path to `proto-gen-grpc-swift`.
+/// - Returns: The command to invoke `protoc` with the `proto-gen-swift` plugin.
+func protocGenSwiftCommand(
+  inputFile: URL,
+  config: GenerationConfig,
+  protoDirectoryPath: URL,
+  protocPath: URL,
+  protocGenSwiftPath: URL
+) throws -> PackagePlugin.Command {
+  let outputPathURL = URL(fileURLWithPath: config.outputPath)
+
+  let outputFilePath = deriveOutputFilePath(
+    for: inputFile,
+    protoDirectoryPath: protoDirectoryPath,
+    outputDirectory: outputPathURL,
+    outputExtension: "pb.swift"
+  )
+
+  let arguments = constructProtocGenSwiftArguments(
+    config: config,
+    using: config.fileNaming,
+    inputFiles: [inputFile],
+    protoDirectoryPaths: [protoDirectoryPath],
+    protocGenSwiftPath: protocGenSwiftPath,
+    outputDirectory: outputPathURL
+  )
+
+  return Command.buildCommand(
+    displayName: "Generating Swift Protobuf files for \(inputFile.relativePath)",
+    executable: protocPath,
+    arguments: arguments,
+    inputFiles: [inputFile, protocGenSwiftPath],
+    outputFiles: [outputFilePath]
+  )
+}
+
+/// Derive the expected output file path to match the behavior of the `proto-gen-swift` and `proto-gen-grpc-swift` `protoc` plugins.
```

Comment:
What behaviour are you trying to match here?

Reply:
Changed to
```
/// Derive the expected output file path to match the behavior of the `protoc-gen-swift` and `protoc-gen-grpc-swift` `protoc` plugins
/// when using the `FullPath` naming scheme.
```